p2p/providers/peerstore_provider/persistence/p2p.go

```diff
@@ -0,0 +1,72 @@
+package persistence
+
+import (
+	"fmt"
+
+	"github.com/pokt-network/pocket/p2p/providers/peerstore_provider"
+	typesP2P "github.com/pokt-network/pocket/p2p/types"
+	"github.com/pokt-network/pocket/shared/modules"
+	"github.com/pokt-network/pocket/shared/modules/base_modules"
+)
+
+var (
+	_ peerstore_provider.PeerstoreProvider = &p2pPeerstoreProvider{}
+	_ p2pPStoreProviderFactory             = &p2pPeerstoreProvider{}
+)
+
+// unstakedPeerstoreProvider is an interface which the p2p module supports in
+// order to allow access to the unstaked-actor-router's peerstore
+//
+// TECHDEBT(#xxx): will become unnecessary after `modules.P2PModule#GetUnstakedPeerstore` is added.`
+// CONSIDERATION: split `PeerstoreProvider` into `StakedPeerstoreProvider` and `UnstakedPeerstoreProvider`.
+// (see: https://github.com/pokt-network/pocket/pull/804#issuecomment-1576531916)
+type unstakedPeerstoreProvider interface {
+	GetUnstakedPeerstore() (typesP2P.Peerstore, error)
+}
+
+type p2pPStoreProviderFactory = modules.Factory[peerstore_provider.PeerstoreProvider]
+
+type p2pPeerstoreProvider struct {
+	base_modules.IntegratableModule
+	persistencePeerstoreProvider
+
+	p2pModule modules.P2PModule
+}
+
+func NewP2PPeerstoreProvider(
```

Comment:
Optional NIT: Personally, I think the multilined method header here makes it harder to read (given only one parameter), but it's such a subjective preference feel free to ignore.